cas-server-core/src/main/java/org/jasig/cas/services/DefaultRegisteredServiceAuthorizationStrategy.java

```diff
@@ -0,0 +1,256 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.services;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This is {@link DefaultRegisteredServiceAuthorizationStrategy}
+ * that allows the following rules:
+ *
+ * <ul>
+ *     <li>A service may be disallowed to use CAS for authentication</li>
+ *     <li>A service may be disallowed to take part in CAS single sign-on such that
+ *     presentation of credentials would always be required.</li>
+ *     <li>A service may be prohibited from receiving a service ticket
+ *     if the existing principal attributes don't contain the required attributes
+ *     that otherwise grant access to the service.</li>
+ * </ul>
+ *
+ * @author Misagh Moayyed mmoayyed@unicon.net
+ * @since 4.1
+ */
+public class DefaultRegisteredServiceAuthorizationStrategy implements RegisteredServiceAuthorizationStrategy {
+
+    private static final long serialVersionUID = 1245279151345635245L;
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    /** Is the service allowed at all? **/
+    private boolean enabled = true;
+
+    /** Is the service allowed to use SSO? **/
+    private boolean ssoEnabled = true;
+
+    /**
+     * Defines the attribute aggregation behavior when checking for required attributes.
+     * Default requires that all attributes be present and match the principal's.
+     */
+    private boolean requireAllAttributes = true;
+
+    /**
+     * Collection of required attributes
+     * for this service to proceed.
+     */
+    private Map<String, Set<String>> requiredAttributes = new HashMap<>();
+
+    /**
+     * Instantiates a new Default registered service authorization strategy.
+     * By default, rules indicate that services are both enabled
+     * and can participate in SSO.
+     */
+    public DefaultRegisteredServiceAuthorizationStrategy() {
+        this(true, true);
+    }
+
+    /**
+     * Instantiates a new Default registered service authorization strategy.
+     *
+     * @param enabled the enabled
+     * @param ssoEnabled the sso enabled
+     */
+    public DefaultRegisteredServiceAuthorizationStrategy(final boolean enabled, final boolean ssoEnabled) {
+        this.enabled = enabled;
+        this.ssoEnabled = ssoEnabled;
+    }
+
+    public final void setEnabled(final boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    /**
+     * Set to enable/authorize this service.
+     * @param ssoEnabled true to enable service
+     */
+    public final void setSsoEnabled(final boolean ssoEnabled) {
```

Comment:
Yes that is a possibility. Doing cascade updates from the old to the new simply from the JPA perspective would be very difficult. It might be easier if we prepped a scripted patch that does the update and release that for people to run after the cas install, or to make explicit in the release notes that schema needs to be looked after.